bin/functions/load-config.py

```diff
@@ -327,152 +307,231 @@ def generate_optional_value():  # get some critical values from environment or m
-                else join(HibenchConf["hibench.hadoop.home"], "etc", "hadoop")
-            HibenchConfRef["hibench.hadoop.configure.dir"] = "Inferred by: 'hibench.hadoop.version' & 'hibench.hadoop.release'"
-        elif HibenchConf["hibench.hadoop.release"].startswith("cdh"): # CDH release
-            HibenchConf["hibench.hadoop.configure.dir"] = join(HibenchConf["hibench.hadoop.home"], "etc", "hadoop-mapreduce1") if HibenchConf["hibench.hadoop.version"] == "hadoop1" \
-                else join(HibenchConf["hibench.hadoop.home"], "etc", "hadoop")
-            HibenchConfRef["hibench.hadoop.configure.dir"] = "Inferred by: 'hibench.hadoop.version' & 'hibench.hadoop.release'"
-
+        # Apache and HDP release
+        if HibenchConf["hibench.hadoop.release"] == "apache" or HibenchConf["hibench.hadoop.release"] == "hdp":
+            HibenchConf["hibench.hadoop.configure.dir"] = join(HibenchConf["hibench.hadoop.home"], "etc", "hadoop")
+            HibenchConfRef["hibench.hadoop.configure.dir"] = "Inferred by: 'hibench.hadoop.release'"
+        # CDH release
+        elif HibenchConf["hibench.hadoop.release"].startswith("cdh"):
+            HibenchConf["hibench.hadoop.configure.dir"] = join(HibenchConf["hibench.hadoop.home"], "etc", "hadoop")
+            HibenchConfRef["hibench.hadoop.configure.dir"] = "Inferred by:  & 'hibench.hadoop.release'"
+
+def probe_mapper_reducer_names():
     # set hadoop mapper/reducer property names
     if not HibenchConf.get("hibench.hadoop.mapper.name", ""):
-        HibenchConf["hibench.hadoop.mapper.name"] = "mapred.map.tasks" if HibenchConf["hibench.hadoop.version"] == "hadoop1" else "mapreduce.job.maps"
-        HibenchConfRef["hibench.hadoop.mapper.name"] = "Inferred by: 'hibench.hadoop.version'"
+        HibenchConf["hibench.hadoop.mapper.name"] = "mapreduce.job.maps"
+        HibenchConfRef["hibench.hadoop.mapper.name"] = "Use default mapper name"
     if not HibenchConf.get("hibench.hadoop.reducer.name", ""):
-        HibenchConf["hibench.hadoop.reducer.name"] = "mapred.reduce.tasks" if HibenchConf["hibench.hadoop.version"] == "hadoop1" else "mapreduce.job.reduces"
-        HibenchConfRef["hibench.hadoop.reducer.name"] = "Inferred by: 'hibench.hadoop.version'"
+        HibenchConf["hibench.hadoop.reducer.name"] = "mapreduce.job.reduces"
+        HibenchConfRef["hibench.hadoop.reducer.name"] = "Use default reducer name"
 
+def probe_masters_slaves_hostnames():
     # probe masters, slaves hostnames
     # determine running mode according to spark master configuration
-    if not (HibenchConf.get("hibench.masters.hostnames", "") or HibenchConf.get("hibench.slaves.hostnames", "")): # no pre-defined hostnames, let's probe
+    if not (HibenchConf.get("hibench.masters.hostnames", "") or HibenchConf.get("hibench.slaves.hostnames",
+                                                                                "")):  # no pre-defined hostnames, let's probe
```

Comment:
Probably not, the two rows have the same length while they follow different styles.
For the newest code of these two lines, there are many line feeds. It seems weird for new python developers such as me, but it accords with pep8